### ISSUE TITLE
ORCIDs are saved into database as whole URI.

### ISSUE THREAD

**Issue by WaDelma** [pkp/pkp-lib#2059]
Currently ORCIDs are asked to be written with whole URI (http://orcid.org/****-****-****-****) and saved as such into database.

orcid.org allows using `https` so the better way to display links to ORCIDs would be: 'https://orcid.org/****-****-****-****'.

I could remove that prefix and substitute it with the `https` version, but the bigger problem is that OJS is saving ORCIDs as URIs in database: Preferred showing conventions can change as happened with DOIs recently.

So I would propose changing it so that only the identification part of ORCIDs would be saved to database. This is annoying change as migration scripts should take it to account and edit rows in the database.

**Comment by asmecher** [comment]
@WaDelma, ORCID seems to require the use of complete, `http` URLs according to its guidelines:
https://orcid.org/trademark-and-id-display-guidelines
I suspect they'll probably move over to `https` at some point, but until they do, I'd like to stick to their standards.

**Comment by WaDelma** [comment]
Well the URL can be constructed in front end and not saved in the database.

It's also kinda strange that OSJ asks the ORCID as URL even though it's not necessary.

**Comment by mtub** [comment]
@asmecher, just adding my two cents here...

The ORCID guidelines cover the display of ORCID identifiers, they do not forbid that we ask for (and store) ORCID iDs in a bare, non-URI format. Also, the guidelines merely recommend

> Our recommended display is a hyperlinked URI 

which I read as protocol-agnostic (though all examples are http and in some older document they explicitly sag "http-URI". My favorite way of handling ORCID iDs would be:

a) let users enter their ORCID iDs in any format they like (http URI, https URI, just the Identifier...)
b) strip protocol/URI from input if necessary
c) validate ORCID iD 
d) store in simple format like @WaDelma suggested
e) display according to their guidelines, with ORCID iD image, active link (e.g. 
https://orcid.org/0000-0002-4905-4662) and shortened URI (orcid.org/0000-0002-4905-4662)

If you feel that it should stay an http URI for the moment, then this would still work and you could change the display behaviour whenever the time feels right.

(and btw: I feel the same about DOIs. it's easier to store just the identifier and to be able to construct displayed values and URIs etc. from that, IMHO)

**Comment by asmecher** [comment]
Hi all -- have a look at this prior Bugzilla entry:
https://pkp.sfu.ca/bugzilla/show_bug.cgi?id=8963

**Comment by WaDelma** [comment]
Okay. I still would prefer saving only the ID part in database so changing it's display wouldn't need database changes.

**Comment by WaDelma** [comment]
Apparently ORCID is working on changing to https, but it will take some time: http://support.orcid.org/forums/175591-orcid-ideas-forum/suggestions/7018328-use-https-by-default
That Trello link in there doesn't seem to work, so can't see how far along they are.

**Comment by asmecher** [comment]
@WaDelma, I suspect they'll update their policies in order to permit `https` URLs, and at that point I agree it might make sense to just store the ID part. I'd just rather not second-guess their plans until then, and for now, they're fixed to `http` URLs. So yes, I think we're probably agreed on approach and reasoning, just perhaps not on timing :)

**Comment by WaDelma** [comment]
Fair enough. I was just kinda shocked that ORCID iDs were in URL form in the database :P

**Comment by WaDelma** [comment]
I contacted ORCID support and got this answer:

> Thanks for contacting ORCID Support!
> 
> The choice to use http instead of https with ORCID identifiers was made when ORCID first launched because it was through the http prefix fit better with ORCID's public, open nature. We agree with you that https would be better, but given the massive effort needed to update how everyone stores and resolves ORCID iDs we have decided not to force that change for now.
> 
> However, I think you raise a good point in that our display guidelines should allow for https. I'll raise that with our communications team and see if we can get those updated.

**Comment by asmecher** [comment]
Thanks, @WaDelma! Please let me know what you hear.

**Comment by mtub** [comment]
Hi @asmecher, since this is still open, please note that the ORCID display guidelines now demand that we use the https links…

https://orcid.org/trademark-and-id-display-guidelines#02-d-i
